crates/builder/src/tasks/tx_poller.rs

```diff
@@ -0,0 +1,110 @@
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use std::{collections::HashMap, time};
+
+use alloy_consensus::TxEnvelope;
+use eyre::Error;
+use reqwest::Client;
+use serde_json::from_str;
+use serde_json::Value;
+use tokio::sync::mpsc;
+use tokio::task::JoinHandle;
+
+pub use crate::config::BuilderConfig;
+
+/// Implements a poller for the block builder to pull transactions from the transaction pool.
+pub struct TxPoller {
+    // config for the builder
+    pub config: BuilderConfig,
+    // Reqwest client for fetching from the tx-pool
+    pub client: Client,
+    //  Maintain a set of txn hash to expiration time
+    pub seen_txns: Arc<Mutex<HashMap<String, time::Instant>>>,
+}
+
+/// TxPoller implements a poller that fetches unique transactions from the transaction pool.
+impl TxPoller {
+    /// returns a new TxPoller with the given config.
+    pub fn new(config: &BuilderConfig) -> Self {
+        Self {
+            config: config.clone(),
+            client: Client::new(),
+            seen_txns: Arc::new(Mutex::new(HashMap::new())),
```

Comment:
if `seen_txns` is not shared with any other task, the arc and mutex are unnecessary. Given that we seem to externalize txns via a channel, i think it is not shared with any other tasks